source/type/message.cpp

```diff
@@ -707,20 +710,23 @@ boost::optional<RailSection> try_make_rail_section(
         if (routes.empty()) {
             routes = line->route_list;
         }
+    }
+
+    if (!routes.empty()) {
         // some routes were given, let's check that they belong to "line"
-        else {
-            for (const Route* route : routes) {
-                if (route->line == nullptr) {
-                    LOG4CPLUS_WARN(logger, "Rail section has line: '"
-                                               << *line_uri << "' but also a route with no line: " << route->uri);
-                    return boost::none;
-                }
-                if (route->line->idx != line->idx) {
-                    LOG4CPLUS_WARN(logger, "Rail section has line: '"
-                                               << *line_uri << "' but also a route: '" << route->uri
-                                               << "' that belongs to a different line: " << route->line->uri);
-                    return boost::none;
-                }
+        for (const Route* route : routes) {
+            if (route->line == nullptr) {
+                LOG4CPLUS_WARN(logger, "Rejected rail section has line: '"
+                                           << *line_uri << "' but also a route with no line: " << route->uri);
+                return boost::none;
+            } else if (line == nullptr) {
+                line = route->line;
+            }
+            if (route->line->idx != line->idx) {
+                LOG4CPLUS_WARN(logger, "Rejected rail section has line (or a route from line): '"
+                                           << *line_uri << "' but also a route: '" << route->uri
+                                           << "' that belongs to a different line: " << route->line->uri);
+                return boost::none;
             }
```

Comment:
```suggestion
            if (route->line == nullptr) {
                LOG4CPLUS_WARN(logger, "Rejected rail section has line: '"
                                           << *line_uri << "' but also a route with no line: " << route->uri);
                return boost::none;
            }
            if (line == nullptr) {
                line = route->line;
            }else if (route->line->idx != line->idx) {
                LOG4CPLUS_WARN(logger, "Rejected rail section has line (or a route from line): '"
                                           << *line_uri << "' but also a route: '" << route->uri
                                           << "' that belongs to a different line: " << route->line->uri);
                return boost::none;
            }
```

Reply:
Done in https://github.com/hove-io/navitia/pull/4073/commits/b808e034ba00a17bdd6afcad1ba559d0bdae81a0